src/exports.cc

```diff
@@ -715,6 +715,29 @@ Word writevImpl(Word fd, Word iovs, Word iovs_len, Word *nwritten_ptr) {
   return 0; // __WASI_ESUCCESS
 }
 
+// __wasi_errno_t path_open(__wasi_fd_t fd, __wasi_lookupflags_t dirflags, const char *path,
+// size_t path_len, __wasi_oflags_t oflags, __wasi_rights_t fs_rights_base, __wasi_rights_t
+// fs_rights_inheriting,  __wasi_fdflags_t fdflags, __wasi_fd_t *retptr0)
+Word wasi_unstable_path_open(Word fd, Word dir_flags, Word path, Word path_len, Word oflags,
+                             int64_t fs_rights_base, int64_t fg_rights_inheriting, Word fd_flags,
+                             Word nwritten_ptr) {
+  return 52; // __WASI_ERRNO_ENOSYS
+}
+
+// __wasi_errno_t __wasi_fd_read(__wasi_fd_t fd, __wasi_prestat_t *retptr0)
+Word wasi_unstable_fd_prestat_get(Word fd, Word buf_ptr) {
+  // If we throw other errors, the preopen in WASI ctors immediately exits the program,
+  // so we just return BADF so that ctors exist successfully even if
+  // we don't support any preopens yet.
+  // https://github.com/WebAssembly/wasi-libc/blob/f2e779e5f1ba4a539937cedeeaa762c1e0c162df/libc-bottom-half/sources/preopens.c#L218-L221
+  return 8; // __WASI_ERRNO_BADF
```

Comment:
`BADF` is the correct return value here, since it's expected to be emitted once we iterate over all preopen directories, so I'm not sure if we need that comment is necessary.